cads_adaptors/adaptors/arco.py

```diff
@@ -36,18 +37,43 @@ def _normalise_location(self, request: Request) -> None:
         except (ValueError, TypeError):
             raise InvalidRequest(f"Invalid {location=}. {msg}")
 
-    def _normalise_date(self, request: Request) -> None:
-        date = ensure_list(request.get("date"))
-        if not date:
-            request["date"] = date
-            return
-
-        if len(date) != 1:
+    def _normalise_date(self, request: Request, date_key="date") -> None:
```

Comment:
I don't think it's a good idea to introduce a new syntax that does the same thing. I would suggest enforcing a single, obvious way to do it. However, if you want or need to proceed, the syntaxes should be consistent with each other (e.g., only one is sorted, only one allows more than two dates, etc.).
I would do something like this.
```python
date = ensure_list(request.get(date_key))
date_range = sorted(date[0].split("/") if len(date) == 1 else date)
if len(date_range) == 1:
    date_range *= 2
if len(date_range) != 2:
    raise
```
So define `date_range` at the very beginning, then apply the same code.

Reply:
Addressed, just in a different way